test/libsolidity/semanticTests/modifiers/modifier_in_constructor_ice.sol

```diff
@@ -0,0 +1,7 @@
+// The IR of this contract used to throw
+contract A { modifier m1{_;} }
+contract B is A {constructor() A() m1{} }
```

Comment:
Is this the minimal example btw?

Reply:
This test was generated from the fuzzer. I'll also add `function_modifier_for_constructor.sol`, since this PR also fixes it.